ion/tokenizer.go

```diff
@@ -1263,3 +1273,24 @@ func (t *tokenizer) unread(c int) {
 	t.pos--
 	t.buffer = append(t.buffer, c)
 }
+
+func isInvalidChar(c int) bool {
+	if c < 0x00 || c > 0x1F {
```

Comment:
I found a couple things easier treating EOF as a character instead of an error, and it's an internal-only API so it felt like a good trade-off. It's very possible that's just my old C habits speaking though. :) Feel free to refactor it if you think it improves readability.

Reply:
:+1: for having this treat -1 as an invalid char. In both places we're calling it we're checking for -1 immediately after, which you could then remove.

Reply:
Modified and created #60 to refactor ```read()```